starlite/parsers.py

```diff
@@ -57,9 +55,8 @@ def parse_url_encoded_form_data(encoded_data: bytes, encoding: str) -> Dict[str,
     """
     decoded_dict: DefaultDict[str, List[Any]] = defaultdict(list)
     for k, v in parse_query_string(query_string=encoded_data, encoding=encoding):
-        if isinstance(v, str):
-            with suppress(JSONDecodeError):
-                v = loads(v)
+        with suppress(JSONDecodeError):
+            v = loads(v) if isinstance(v, str) else v
```

Comment:
@jcrist - this is the one unsupported use we have, which we need to use the stdlib json for. This parses `urlencoded` request bodies.

Values can be stringified lists: `"[1,2,3,4]"` and stringified numbers / floats, which are converted into their respective data types.

Reply:
Thanks for the ping. Wouldn't the `starlite.utils.decode_json` method work here? The overhead of decoding from a string here should be dwarfed by the performance of upstream processing code like `parse_qsl`. These strings are also likely to be short, so the unnecessary copy when decoding from a utf-8 string shouldn't matter as much.

Reply:
No, this raises a `TypeError` from msgspec. 

Of course this is not the main performance issue, but we have no choice but to do both in this case, that or start using regex, which can be a problem.

Reply:
Looks like it's working to me?

```python
In [8]: decode_json("[1, 2, 3, 4, 5]")  # starlite.utils.serialization.decode_json
Out[8]: [1, 2, 3, 4, 5]

In [9]: decode_json("1")
Out[9]: 1
```

Reply:
yup, i wonder why it did work for me before.